js/ui/pivot_grid/ui.pivot_grid.export.js

```diff
@@ -255,34 +253,45 @@ export const DataProvider = Class.inherit({
         return result;
     },
 
-    _tryGetAreaName(items, item, rowIndex, cellIndex) {
-        const columnHeaderSize = items[0][0].rowspan;
-        const rowHeaderSize = items[0][0].colspan;
-
-        if(cellIndex >= rowHeaderSize && rowIndex < columnHeaderSize) {
+    _tryGetAreaName(item, rowIndex, cellIndex) {
+        if(this.isColumnAreaCell(rowIndex, cellIndex)) {
             return 'column';
-        } else if(rowIndex >= columnHeaderSize && cellIndex < rowHeaderSize) {
+        } else if(this.isRowAreaCell(rowIndex, cellIndex)) {
             return 'row';
         } else if(isDefined(item.dataIndex)) {
             return 'data';
         }
     },
 
+    isRowAreaCell(rowIndex, cellIndex) {
+        return rowIndex >= this.getColumnAreaSize() && cellIndex < this.getRowAreaSize();
+    },
+
+    isColumnAreaCell(rowIndex, cellIndex) {
+        return cellIndex >= this.getRowAreaSize() && rowIndex < this.getColumnAreaSize();
+    },
+
+    getColumnAreaSize() {
```

Comment:
imho getColumnAreaRowCount is better, function returns number of rows in columns area (columns header?)

Reply:
Sounds good. Fixed in 9de59b1